code/controllers/subsystems/trade.dm

```diff
@@ -2,16 +2,21 @@ SUBSYSTEM_DEF(trade)
 	name = "Trade"
 	wait = 1 MINUTE
 	priority = SS_PRIORITY_TRADE
-	//Initializes at default time
+	init_order = SS_INIT_MISC_LATE
 
 	var/list/traders = list()
 	var/tmp/list/current_traders
 	var/max_traders = 10
 
 /datum/controller/subsystem/trade/Initialize()
 	. = ..()
-	for(var/i in 1 to rand(1,3))
-		generate_trader(1)
+
+	for(var/trader_type in GLOB.using_map.get_initial_traders())
+		traders += new trader_type
+	var/total_traders = rand(1,3) - length(traders)
```

Comment:
only unsure on this limitation, because wouldn't this make trader list basically static for tradeship (cause it starts with 4 traders)

Reply:
This is just the starting traders, my understanding is that they cycle.

Reply:
They do come and go. On bay it's calibrated so that they stay a while, and there are usually close to the maximum number always present; can recalibrate if you want.